app/src/main/java/org/dicio/dicio_android/eval/SkillEvaluator.java

```diff
@@ -344,22 +346,31 @@ public void afterTextChanged(final Editable s) {
     // Skill evaluation //
     //////////////////////
 
-    private void evaluateMatchingSkill(final String input) {
+    private void evaluateMatchingSkill(final List<String> inputs) {
         if (evaluationDisposable != null && !evaluationDisposable.isDisposed()) {
             evaluationDisposable.dispose();
         }
 
         evaluationDisposable = Single.fromCallable(() -> {
-            final List<String> inputWords = WordExtractor.extractWords(input);
-            final List<String> normalizedWordKeys = WordExtractor.normalizeWords(inputWords);
-            final Skill skill = skillRanker.getBest(input, inputWords, normalizedWordKeys);
+            Skill skill=null;
+            String input="";
+            for(int i=0;i<inputs.size();i++){
+                input=inputs.get(i);
+                final List<String> inputWords = WordExtractor.extractWords(input);
+                final List<String> normalizedWordKeys = WordExtractor.normalizeWords(inputWords);
+                skill = skillRanker.getBest(input, inputWords, normalizedWordKeys);
+                if (skill != null&&!(skill instanceof TextFallback)) {
+                    break;
+                }
```

Comment:
There is a problem here: what if in the future there could many types of possible fallback skills? This `if` would have to check for them all. As an alternative I would do this:
- modify `skillRanker.getBest()` so that it returns `null` when no skill is good enough for the provided input, instead of returning the feedback skill, so that this `if` can be simplified to just `if (skill != null)`
- add this function to `SkillRanker`:
```java
getFallbackSkill(final String input,
        final List<String> inputWords,
        final List<String> normalizedWordKeys) {
    fallbackSkill.setInput(input, inputWords, normalizedWordKeys);
    return fallbackSkill;
}
```
- below this `for` loop, add this code:
```java
if (skill == null) {
    // no skill was good enough for any of the inputs, use fallback skill
    input = inputs.get(i);
    final List<String> inputWords = WordExtractor.extractWords(input);
    final List<String> normalizedWordKeys = WordExtractor.normalizeWords(inputWords);
    skill = skillRanker.getFallbackSkill(input, inputWords, normalizedWordKeys);
}
```

Reply:
The SkillrankerTest class may give a null-pointer exception, when skillranker.getBest returns null.
Maybe it's better to check if FallbackSkill interface is implemented.

Reply:
> The SkillrankerTest class may give a null-pointer exception, when skillranker.getBest returns null.

That's normal, the test should be changed accordingly

> Maybe it's better to check if FallbackSkill interface is implemented.

Well no, since if the FallbackSkill returned by SkillRankerTest is never used, why return it just to indicate that nothing was found? It's better to return null and handle the FallbackSkill in the caller